Feat/ephemeral storage limit

clementnuss opened the pull request:
add a new ephemeral storage limit constraint, heavily inspired by the `containerlimits` constraint.

> Requires containers to have an ephemeral storage limit set and constrains the limit to be within the specified maximum values

ritazh commented:
Thank you for the PR @clementnuss! Apologies for the delay. Can you PTAL https://open-policy-agent.github.io/gatekeeper-library/website/#development to run few commands as we have updated few things to autogenerate changes for the website and artifacthub.

clementnuss commented:
hi @ritazh 

I've updated my branch by following the latest instructions. can you review this PR ?

clementnuss commented:
@maxsmythe @ritazh 

I think this PR would be beneficial to the gatekeeper library, could it be merged? I'ts been on hold for quite a long time now 🤔

maxsmythe commented on library/general/ephemeralstoragelimit/suite.yaml:
```diff
@@ -0,0 +1,21 @@
+  constraint: samples/container-must-have-ephemeral-storage-limit/constraint.yaml
+  cases:
+  - name: ephemeral-storage-limit-100Mi
+    object: samples/container-must-have-ephemeral-storage-limit/example_allowed_ephemeral-storage.yaml
```
Could we add in a test case for init containers?

clementnuss replied:
sure, good idea!

maxsmythe reviewed (commented):
Sorry this slipped through the cracks. One request for an added test to suite.yaml, other than that, LGTM!

clementnuss commented:
I've added 2 test cases for initContainer: 1 for allowed initContainer and another one with a 1Pi ephemeral storage limit for disallow 👍🏻

maxsmythe reviewed (approved):
Thanks!

LGTM

sozercan reviewed (approved):
Sorry about the delay and thank you! LGTM pending tests

clementnuss commented:
hey @sozercan 

Could you let the CI pipeline run again ? I've updated website docs and artifactshub artifacts as I had forgotten to update that when I added the two tests.

pending tests
